profile/src/v100/app_preferences/app_preferences.rs

```diff
@@ -0,0 +1,239 @@
+use crate::prelude::*;
+
+/// Collection of all settings, preferences and configuration related to how the wallet
+/// behaves and looks.
+///
+/// Current and other saved Gateways, security settings, connected P2P clients,
+/// App Display settings and preferences for transaction.
+#[derive(
+    Debug,
+    Default,
+    Deserialize,
+    Serialize,
+    PartialEq,
+    Eq,
+    Clone,
+    Hash,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+pub struct AppPreferences {
+    /// Display settings in the wallet app, such as appearances, currency etc.
+    pub display: AppDisplay,
+
+    /// The gateway of the active network and collection of other saved gateways.
+    pub gateways: Gateways,
+
+    /// Collection of clients user have connected P2P with, typically these
+    /// are WebRTC connections with DApps, but might be Android or iPhone
+    /// clients as well.
+    pub p2p_links: P2PLinks,
+
+    /// Controls e.g. if Profile Snapshot gets synced to iCloud/Google backup or not.
+    pub security: Security,
+
+    /// Default config related to making of transactions
+    pub transaction: Transaction,
```

Comment:
we might wanna call `Transaction` something else, perhaps `AppPreferencesTransaction` (maybe prefix all the types used by this type with that...?)